templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -488,7 +488,7 @@
      });
 
      function showHideCancelInfo(obj) {
-       if (obj.find(":selected").text() == 'Refunded' || obj.find(":selected").text() == 'Cancelled') {
+       if (obj.find(":selected").val() == '7' || obj.find(":selected").val() == '3') {
```

Comment:
So the question is are you willing/able to pursue that additional bit or would you prefer I merge as-is?